File: inject-java/src/test/groovy/io/micronaut/inject/configproperties/MyConfigInner.java
package io.micronaut.inject.configproperties;

import io.micronaut.context.annotation.ConfigurationProperties;

import java.util.List;

@ConfigurationProperties("foo.bar")
public class MyConfigInner {

    private List<InnerVal> innerVals;

    public List<InnerVal> getInnerVals() {
        return innerVals;
    }

    public void setInnerVals(List<InnerVal> innerVals) {
        this.innerVals = innerVals;
    }

    public static class InnerVal {

        private Integer expireUnsignedSeconds;

        public Integer getExpireUnsignedSeconds() {
            return expireUnsignedSeconds;
        }

        public void setExpireUnsignedSeconds(Integer expireUnsignedSeconds) {
            this.expireUnsignedSeconds = expireUnsignedSeconds;
        }
    }

}

File: inject-kotlin/src/test/kotlin/io/micronaut/kotlin/processing/beans/configproperties/MyConfigInner.kt
package io.micronaut.kotlin.processing.beans.configproperties

import io.micronaut.context.annotation.ConfigurationProperties

@ConfigurationProperties("foo.bar")
class MyConfigInner {
    var innerVals: List<InnerVal>? = null

    class InnerVal {
        var expireUnsignedSeconds: Int? = null
    }
}
